java/src/jni/h5aImp.c

```diff
@@ -156,45 +156,61 @@ JNIEXPORT jint JNICALL
 Java_hdf_hdf5lib_H5_H5Aread(JNIEnv *env, jclass clss, jlong attr_id, jlong mem_type_id, jbyteArray buf,
                             jboolean isCriticalPinning)
 {
-    jboolean readBufIsCopy;
-    jbyte   *readBuf = NULL;
-    hsize_t  dims[H5S_MAX_RANK];
-    hid_t    sid = H5I_INVALID_HID;
-    jsize    n;
-    htri_t   vl_data_class;
-    herr_t   status = FAIL;
+    jboolean    readBufIsCopy;
+    jbyte      *readBuf = NULL;
+    hsize_t     dims[H5S_MAX_RANK];
+    hid_t       sid = H5I_INVALID_HID;
+    size_t      typeSize;
+    H5T_class_t type_class;
+    jsize       n;
+    htri_t      vl_data_class;
+    herr_t      status = FAIL;
 
     UNUSED(clss);
 
     if (NULL == buf)
         H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5Aread: read buffer is NULL");
 
+    /* Get size of data array */
+    if ((n = ENVPTR->GetArrayLength(ENVONLY, buf)) < 0) {
```

Comment:
Shouldn't this stay inside the `vl_data_class` block? Looks like it's only used in those blocks.

Reply:
I had to split the vl_data_class in two parts and each needed the number.

Reply:
I guess I could have duplicated it.

Reply:
Each part can still use the number. The variable itself is in the scope of all the blocks, but you can still only set it in the first block.